Documentation/NetworkBoot.md

```diff
@@ -131,13 +167,15 @@ LABEL SerenityOS
         APPEND serenity/prekernel root=/dev/ramdisk0 --- serenity/kernel --- serenity/ramdisk
 ```
 
-### Troubleshooting
+# Troubleshooting
+
+It is recommended to netboot a mainstream operating system on the target at least once before attempting to netboot SerenityOS, otherwise it may be difficult to isolate an issue with SerenityOS from an issue with your network booting setup.
 
-- Issues with DHCP or TFTP usually require sniffing packets on the network to figure out.
-- TFTP is a slow protocol, transferring the QEMU disk image (~ 200 MiB) will take some time. Consider setting up a FTP or HTTP server for faster downloading of SerenityOS artefacts if your bootloader supports it.
-- Remember that SerenityOS has not been extensively tested on physical hardware.
-- Some BIOS implementations of PXE are buggy or some machines may not have a PXE boot option at all in which case you could try using [iPXE](https://ipxe.org/).
-- Virtual machines can also be booted over the network. Cheat notes for QEMU on Linux, assuming `br0` is already set up:
+   - Ensure your paths on the TFTP server and the DHCP options are correct.
+   - Check that your TFTP server is properly working with a TFTP client. Wrong file permissions or ownership on `/srv/tftp` can prevent the TFTP server from functioning properly.
+   - TFTP is a slow protocol, big files can take tens of seconds or even minutes before the transfer finishes. Consider setting up a FTP or HTTP server for faster downloading of SerenityOS artefacts if your bootloader supports it.
+   - Use iPXE if you suspect your network card's built-in PXE firmware is buggy.
```

Comment:
To be honest, iPXE is much more superior to using GRUB but that's my opinion. I tried to use GRUB on my ICH7 test machine and it simply failed due to a bug in the PXE firmware... iPXE however worked like a charm - you just need a supported card (it even supports Atheros WiFi cards so you can authenticate wirelessly!).

Reply:
The only drawback of iPXE is that it cannot set a framebuffer for multiboot kernels. So you will boot into text mode (with the ugly VGA 80x25 resolution). Ofc, this could be fixed with a patch to iPXE :)

Reply:
Here, I'm advising iPXE as a drop-in replacement for a buggy network card PXE firmware. Using iPXE as a bootloader is documented in its own section.